Python/Product/PythonTools/PythonTools/WebProject/PythonWebLauncher.cs

```diff
@@ -134,29 +140,41 @@ LaunchConfiguration defaultConfig
         }
 
 
-        private void StartBrowser(string url, Func<bool> shortCircuitPredicate) {
+        private Task StartBrowser(string url, Func<bool> shortCircuitPredicate) {
             Uri uri;
             if (!String.IsNullOrWhiteSpace(url) && Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out uri)) {
+                var tcs = new TaskCompletionSource<object>();
+
                 OnPortOpenedHandler.CreateHandler(
                     uri.Port,
                     shortCircuitPredicate: shortCircuitPredicate,
                     action: () => {
-                        var web = _serviceProvider.GetService(typeof(SVsWebBrowsingService)) as IVsWebBrowsingService;
-                        if (web == null) {
-                            PythonToolsPackage.OpenWebBrowser(url);
-                            return;
+                        try {
```

Comment:
Spent ten minutes on it and it doesn't just drop in nicely, so we can come back to it later. (It also only seems to benefit IE if I'm reading correctly?)